symengine/test_visitors.cpp

```diff
@@ -754,4 +754,97 @@ tribool is_irrational(const Basic &b)
     RationalVisitor visitor(false);
     return visitor.apply(b);
 }
+
+void AlgebraicVisitor::error()
+{
+    throw SymEngineException(
+        "Only numeric types allowed for is_algebraic/is_transcendental");
+}
+
+void AlgebraicVisitor::bvisit(const Basic &x)
+{
+    is_algebraic_ = tribool::indeterminate;
+}
+
+void AlgebraicVisitor::bvisit(const Set &x)
+{
+    error();
+}
+
+void AlgebraicVisitor::bvisit(const Relational &x)
+{
+    error();
+}
+
+void AlgebraicVisitor::bvisit(const Boolean &x)
+{
+    error();
+}
+
+void AlgebraicVisitor::bvisit(const Add &x)
+{
+    // algebraic + algebraic = algebraic
+    for (const auto &arg : x.get_args()) {
+        arg->accept(*this);
+        if (not is_true(is_algebraic_)) {
+            is_algebraic_ = tribool::indeterminate;
```

Comment:
This become indeterminate only if there are two numbers which are not algebraic. If there's only one number that is not algebraic, it's false.

Reply:
Good catch!  algebraic + transcendental = transcendental

Reply:
`Add` should be good now